### PR TITLE
fix: Document the Destination struct

### PR DESCRIPTION
**Pull request by technovangelist**
Signed-off-by: Matt Williams <m@technovangelist.com>

## Summary

When openapi generates the json document describing the api, a lot of stuff is missing. This pr fixes whats missing from the Destination object.

## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

#3664

### CODE REVIEW

File: api/destination.go [hunk 1]
```diff
@@ -6,21 +6,21 @@ import (
 )
 
 type Destination struct {
-	ID         uid.ID                `json:"id"`
-	UniqueID   string                `json:"uniqueID" form:"uniqueID" example:"94c2c570a20311180ec325fd56"`
-	Name       string                `json:"name" form:"name"`
-	Kind       string                `json:"kind"`
+	ID         uid.ID                `json:"id" note:"ID of the destination" example:"7a0b06b30F"`
```
Comment (ssoroka) [inline api/destination.go:9]: this example isn't a valid ID, there are no `0` in the id set. has to be 11 characters from the set of `123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ`

File: api/destination.go [hunk 2]
```diff
@@ -6,21 +6,21 @@ import (
...
+	UniqueID   string                `json:"uniqueID" form:"uniqueID" note:"Unique ID generated by the connector" example:"94c2c570a20311180ec325fd56"`
+	Name       string                `json:"name" form:"name" note:"Name of the destination" example:"production-cluster"`
+	Kind       string                `json:"kind" note:"Kind of destination. eg. kubernetes or ssh or postgres" example:"kubernetes"`
 	Created    Time                  `json:"created"`
 	Updated    Time                  `json:"updated"`
-	Connection DestinationConnection `json:"connection"`
+	Connection DestinationConnection `json:"connection" note:"Object that includes the URL and CA for the destination" example:"{'url':'localhost:8443', 'ca':'clustercertificate'}"`
```
Comment (ssoroka) [inline api/destination.go:15]: this example should probably be covered by the DestinationConnection struct

**Review by ssoroka — CHANGES_REQUESTED** [review]

File: api/destination.go (hunk 1 above)
Comment (technovangelist) [reply api/destination.go:9]: does that mean the format for the ID is wrong? that shows up as "[\\da-zA-HJ-NP-Z]{1,11}". i think \d means 0-9 not 1-9

File: api/destination.go (hunk 2 above)
Comment (technovangelist) [reply api/destination.go:15]: that doesn't show up in the openapi json doc

File: api/destination.go (hunk 1 above)
Comment (technovangelist) [reply api/destination.go]: but updated the string to use other digits

Comment (technovangelist) [reply api/destination.go]: and pr #3671 fixes it

**Review by BruceMacD — COMMENTED** [review]
Looks good, will approve once the docs are re-generated.

**Review by dnephin — APPROVED** [review]
LGTM!

**Review by jmorganca — APPROVED** [review]

**Pull request merged**